File: pspring-scala4/src/test/scala/com/stulsoft/pspring/scala/SomeServiceTest.scala
/*
 * Copyright (c) 2018. Yuriy Stul
 */

package com.stulsoft.pspring.scala

import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{BeforeAndAfterEach, FunSuite, Matchers, Suite}
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.{ComponentScan, Configuration}
import org.springframework.test.context.support.AnnotationConfigContextLoader
import org.springframework.test.context.{ContextConfiguration, TestContextManager}

/**
  * @author Yuriy Stul
  */
@RunWith(classOf[JUnitRunner])
class SomeServiceTest extends FunSuite with SpringSomeServiceTest with Matchers {
  @Autowired
  val ss: SomeService = null

  test("testUseF1") {
    ss.useF1 should be("implementation 1")
  }

}

@ContextConfiguration(
  classes = Array(classOf[ProdSomeServiceConfig]),
  loader = classOf[AnnotationConfigContextLoader])
trait SpringSomeServiceTest extends BeforeAndAfterEach {
  this: Suite =>

  override def beforeEach(): Unit = {
    new TestContextManager(classOf[SpringSomeServiceTest]).prepareTestInstance(this)
    super.beforeEach()
  }
}

@ComponentScan(basePackages = Array("com.stulsoft.pspring.scala"))
@Configuration
class ProdSomeServiceConfig {
}

File: conf-yml/src/test/java/com/stulsoft/pspring/conf/yml/SomeServiceTest.java
/*
 * Copyright (c) 2020. Yuriy Stul
 */

package com.stulsoft.pspring.conf.yml;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.ConfigFileApplicationContextInitializer;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.FilterType;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.TestPropertySource;
import org.springframework.test.context.support.AnnotationConfigContextLoader;

/**
 * @author Yuriy Stul
 */
@SpringBootTest
@ContextConfiguration(loader = AnnotationConfigContextLoader.class,
        initializers = ConfigFileApplicationContextInitializer.class)
@ComponentScan(basePackages = "com.stulsoft.pspring.conf.yml",
        excludeFilters = @ComponentScan.Filter(value = StarterApplication.class, type = FilterType.ANNOTATION))
@ActiveProfiles(profiles = "test")
@TestPropertySource(locations = "classpath:application-test.yml")
public class SomeServiceTest {
    @Autowired
    private SomeService someService;

    @Test
    void showProps() {
        someService.showProps();
    }
}